stdlib/task/src/main/ballerina/task/Module.md

```diff
@@ -1,178 +1,165 @@
+  
+Following code snippet shows how to create an Appointment using `task:Listener`. It will trigger the `onTrigger()` resource by every two seconds.
 
+```ballerina
+    task:TimerConfiguration timerConfiguration = {
+            interval: 1000,
+            delay: 5000,
+            noOfRecurrences: 3
+    };
+    
+    listener task:Listener appointment = new({
+        appointmentDetails: "0/2 * * * * ?"
+    });
+    service appointmentService on appointment {
+        resource function onTrigger() {
+            io:println("Triggering");
+        }
+    }
 ```
 
-### Task appointments
-
-A task appointment is similar to a real-world appointment. The task appointment is configured to run at a given time 
-pattern. A cron expression is used to define the time, and the frequency a task appointment needs to run. 
+### Task Scheduler
 
-The `onTriggerFunction` function of the task is called when the appointment is due.  If an error is returned when 
-executing the `onTriggerFunction` function, the `onErrorFunction` is called.
+A task `Scheduler` can be used to create timers / appointments dynamically. Then a service can be attached to the `Scheduler`. Scheduler can also be created as a timer or an appointment. The configurations are as same as the `Listener`.
 
-The example given below triggers the `onTrigger` function every 5 seconds. If an error is returned, the `cleanupError` 
-function is called.
+Code snippet for an example timer `task:Scheduler`. 
 
 ```ballerina
-    (function() returns error?) onTriggerFunction = onTrigger;
-    (function (error)) onErrorFunction = cleanupError;
-    app = new task:Appointment(onTriggerFunction, onErrorFunction, "0/05 * * * * ?");
-    app.schedule();
+public function createTimer(int interval, int delay, int recurrences) {
+    task:TimerConfiguration timerConfiguration = {
+            interval: interval,
+            delay: delay,
+            noOfRecurrences: recurrences
+    };
+    task:Scheduler timer = new(timerConfiguration);
+    
+    _ = timer.attach(timerService);
+    _ = timer.start();
+}
+
+service timerService = service {
+    resource function onTrigger() {
+        // Task to run when the timer triggers.
+    }
+};
 ```
 
+`Scheduler` can also be used to create appointments. `Scheduler` has following APIs.
+
+- `start()` - Starts the task scheduler, and run the services attached to it.
+- `stop()` - Stops the task. This will shutdown all the process scheduled on the scheduler.
+- `pause()` - Pause the scheduler. This will temporary halts the task execution.
+- `resume()` - Used to resume the task scheduler, when paused.
+- `attach()` - Will attach a service to the scheduler. An optional parameter, `attachment` can be passed to the function, so that it will propagate into the resource.
+- `detach()` - Detach any attached services from the task.
```

Comment:
Shall we follow a single convention? If we are writing as 'Starts the task scheduler', Let's follow the same way in every other place as well.

EX:  `resume()` - resumes ...
`attach()` - attaches ...